File: DSA/vaishnavi_joshi_590016947/Day22/Q2.java
import java.util.*;

public class Main {

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);

        int n = sc.nextInt(); // Number of cards
        int[] cards = new int[n];

        for (int i = 0; i < n; i++) {
            cards[i] = sc.nextInt();
        }

        int left = 0;
        int right = n - 1;
        int player1 = 0;
        int player2 = 0;
        boolean turn = true; // true = Player1, false = Player2

        while (left <= right) {
            int picked;

            if (cards[left] > cards[right]) {
                picked = cards[left];
                left++;
            } else {
                picked = cards[right];
                right--;
            }

            if (turn) {
                player1 += picked;
            } else {
                player2 += picked;
            }

            turn = !turn;
        }

        System.out.println(player1 + " " + player2);

        sc.close();
    }
}

File: DSA/Aarush_590017879/Day_22/Q2.cpp
#include <iostream>
#include <vector>
using namespace std;

int main() {
    int n;
    cin >> n;

    vector<int> cards(n);

    for (int i = 0; i < n; i++) {
        cin >> cards[i];
    }

    int left = 0, right = n - 1;
    int player1 = 0, player2 = 0;
    bool turn = true;

    while (left <= right) {
        int chosen;

        if (cards[left] >= cards[right]) {
            chosen = cards[left];
            left++;
        } else {
            chosen = cards[right];
            right--;
        }

        if (turn)
            player1 += chosen;
        else
            player2 += chosen;

        turn = !turn;
    }

    cout << player1 << " " << player2 << endl;

    return 0;
}